proto/sentry_protos/snuba/v1/endpoint_aggregate_bucket.proto

```diff
@@ -0,0 +1,51 @@
+syntax = "proto3";
+
+package sentry_protos.snuba.v1;
+
+import "sentry_protos/snuba/v1/request_common.proto";
+import "sentry_protos/snuba/v1/trace_item_attribute.proto";
+import "sentry_protos/snuba/v1/trace_item_filter.proto";
+
+enum Function {
+  FUNCTION_UNSPECIFIED = 0;
+  FUNCTION_SUM = 1;
+  FUNCTION_AVERAGE = 2;
+  FUNCTION_COUNT = 3;
+  FUNCTION_P50 = 4;
+  FUNCTION_P90 = 5;
+  FUNCTION_P95 = 6;
+  FUNCTION_P99 = 7;
+  FUNCTION_AVG = 8;
+  FUNCTION_MAX = 9;
+  FUNCTION_MIN = 10;
+  FUNCTION_UNIQ = 11;
+}
+
+message AggregateBucketRequest {
+  RequestMeta meta = 1;
+  TraceItemFilter filter = 2;
+  repeated AttributeAggregation aggregations = 3;
+  uint64 granularity_secs = 4;
+  repeated AttributeKey group_by = 5;
+}
+
+message AttributeAggregation {
+  Function aggregate = 1;
+  AttributeKey key = 2;
+  string label = 3;
+}
+
+message TimeSeries {
+  string label = 1;
+  // the names of the attribute keys which
+  // were in the group by
+  repeated string attribute_names = 2;
```

Comment:
I am a little confused about how this works with group by.

In theory there would be one timeseries for each unique combination of attribute values across all the attribute keys in the GROUP BY.

Does this field correspond to those values? And if so, how can these values be tied back to the corresponding keys?

Reply:
yes you are correct, this needs to be a map<str, str> for attribute names and values